packages/client/hmi-client/src/components/workflow/ops/dataset/tera-dataset-node.vue

```diff
@@ -99,18 +98,15 @@ async function getDatasetById(id: string) {
 
 	if (dataset.value && dataset.value?.id) {
 		// Once a dataset is selected the output is assigned here,
-		// if there is already an output do not reassign
-		if (isEmpty(props.node.outputs)) {
-			emit('update-state', {
-				datasetId: dataset.value.id
-			});
-
-			emit('append-output', {
-				type: 'datasetId',
-				label: dataset.value.name,
-				value: [dataset.value.id]
-			});
-		}
+		emit('update-state', {
+			datasetId: dataset.value.id
+		});
+
+		emit('append-output', {
+			type: 'datasetId',
+			label: dataset.value.name,
+			value: [dataset.value.id]
+		});
```

Comment:
Now everytime this node mounts a new output is created. That if statement from before prevented that. Now you gotta check if the first output value is null before you add the output.

Reply:
Good catch!